src/Mod/Sketcher/Gui/CommandSketcherTools.cpp

```diff
@@ -983,50 +983,61 @@ void CmdSketcherRestoreInternalAlignmentGeometry::activated(int iMsg)
 
     getSelection().clearSelection();
 
-    // go through the selected subelements
-    for (std::vector<std::string>::const_iterator it=SubNames.begin(); it != SubNames.end(); ++it) {
-        // only handle edges
-        if ((it->size() > 4 && it->substr(0,4) == "Edge") ||
-            (it->size() > 12 && it->substr(0,12) == "ExternalEdge")) {
-            int GeoId;
-            if (it->substr(0,4) == "Edge")
-               GeoId = std::atoi(it->substr(4,4000).c_str()) - 1;
-            else
-               GeoId = -std::atoi(it->substr(12,4000).c_str()) - 2;
-
-            const Part::Geometry *geo = Obj->getGeometry(GeoId);
-            // Only for supported types
-            if (geo->getTypeId() == Part::GeomEllipse::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomArcOfEllipse::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomArcOfHyperbola::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomArcOfParabola::getClassTypeId() ||
-                geo->getTypeId() == Part::GeomBSplineCurve::getClassTypeId()) {
+    std::vector<int> SubGeoIds(SubNames.size());
```

Comment:
As you later do the elements unique and sort them "reversed", that would be probably be more efficient to use directly a `std::set` with `std::greater` as a compare operator.

Reply:
I'm not sure I know how to do that. If you have a snippet I could try it out. I don't expect much difference in the two approaches but with set it might be cleaner.

As such far too often I expect only one element will be selected.

Reply:
Here it is : https://github.com/0penBrain/FreeCAD/commit/43f948ac0f2f40f6e96226fc04376dd5a1f4e171

Sidenote : `std::erase_if` has only been introduced in C++20 so now using a check at beginning of for loop to keep compatibility with C++17

Reply:
That looks clean. if you want you can make a PR into my branch.

Reply:
Done